src/mewbot/io/rss.py

```diff
@@ -0,0 +1,514 @@
+        """
+        polling_interval is the time that the Input should wait between polling a new site.
+        The total time between repoll of each site is polling_every.
+        """
+        # If we have no sites to poll, then we shall do nothing
+        if len(self._sites) == 0:
+            return None
+        return float(self._polling_every) / float(len(self._sites))
+
+    @property
+    def sites(self) -> List[str]:
+        return self._sites
+
+    @sites.setter
+    def sites(self, new_sites: List[str]) -> None:
+        self._sites = new_sites
+        self._sites_iter = cycle(iter(self._sites))
+
+    @staticmethod
+    def produces_inputs() -> Set[Type[InputEvent]]:
+        """
+        Defines the set of input events this Input class can produce.
+        """
+        return {
+            RSSInputEvent,
+        }
+
+    @property
+    def polling_every(self) -> int:
+        return self._polling_every
+
+    @polling_every.setter
+    def polling_every(self, new_polling_time: int) -> None:
+        self._polling_every = new_polling_time
+
+    def _get_entry_uid(self, entry: feedparser.util.FeedParserDict, site_url: str) -> str:
+        """
+        Returns a unique id (on a system level) for the given entry and site combination.
+        Used to determine if entries have been put on the wire or not
+        """
+        try:
+            entry_title = entry.title
+        except AttributeError:
+            entry_title = TITLE_NOT_SET_STR
+
+        try:
+            entry_guid = entry.guid
+        except AttributeError:
+            self._logger.warning(
+                "entry %s did not possess a guid - %s might be producing malformed RSS entries",
+                entry,
+                site_url,
+            )
+
+            # This is not a fantastic fallback - but it should work most of the time
+            entry_guid = f"{entry_title}"
+
+        return f"{entry_guid}:{site_url}"
+
+    async def run(self) -> None:
+        """
+        Fires up an aiohttp app to run the service.
+        Token needs to be set by this point.
+        """
+        self._logger.info(
+            "About to start RSS polling - %s will be polled every %s seconds",
+            self._sites,
+            self._polling_every,
+        )
+
+        for target_site in self._sites_iter:
+
+            if target_site not in self.sites_started:
+                asyncio.get_running_loop().create_task(self.startup_site_feed(target_site))
+                # Delay before getting the next site
+                # stagger site reads to prevent overwealming anything
+                await asyncio.sleep(self.polling_every)
+                continue
+
+            asyncio.get_running_loop().create_task(self.poll_site_feed(target_site))
+            # Delay before getting the next site
+            # stagger site reads to prevent overwealming anything
+            await asyncio.sleep(self.polling_every)
```

Comment:
I'm not sure if it improves readability, but maybe

```suggestion
        for target_site in self._sites_iter:

            if target_site not in self.sites_started:
                future = self.startup_site_feed(target_site)
            else:
                future = self.poll_site_feed(target_site)

            asyncio.get_running_loop().create_task(future)
            # Delay before getting the next site
            # stagger site reads to prevent overwealming anything
            await asyncio.sleep(self.polling_every)
```

Reply:
`asyncio.get_running_loop()` could even be determined at `__init__` time, to be honest

Reply:
I meant to say "I agree with this, it removes an implicit branch and also avoids a weird branch to look at in the code" but SOMEONE misread "Commit Suggestion" as "Comment on suggestion"
(it was me, I misread github)

Reply:
ANYWAY
line 255 here has an errant ). Any change would want to remove it

Reply:
I sneakily fixed the suggestion at some point :3

Reply:
> `asyncio.get_running_loop()` could even be determined at `__init__` time, to be honest

Loop isn't up at __init__ - but it can wrap it as a property - which _might_ improve efficiency somewhat.

Reply:
Updated to improve readability